common/common.go

```diff
@@ -96,14 +103,15 @@ func getBlockFromRPC(rpcClient *rpcclient.Client, height int) (*walletrpc.Compac
 	return block.ToCompact(), nil
 }
 
-func BlockIngestor(rpcClient *rpcclient.Client, cache *BlockCache, log *logrus.Entry, startHeight int) {
+// The repetition count, rep, is nonzero only for unit-testing
+func BlockIngestor(rf RpcFunc, sf SleepFunc, cache *BlockCache, log *logrus.Entry, startHeight int, rep int) {
 	reorgCount := 0
 	height := startHeight
 
 	// Start listening for new blocks
 	retryCount := 0
-	for {
-		block, err := getBlockFromRPC(rpcClient, height)
+	for i := 0; rep == 0 || i < rep; i++ {
```

Comment:
The `rep` variable prevents this function from infinite-looping during unit testing (it will, correctly, infinite loop during production).